app/models/concerns/import_job.rb

```diff
@@ -35,7 +35,19 @@ def validate_headers(filename)
 
     log("Headers in file: #{headers}", :debug)
     log("Valid headers for model: #{valid_headers}", :debug)
-    valid_headers == headers
+
+    success = valid_headers == headers
+    unless success
+      ActionCable.server.broadcast "import_job", { html:
+        "<div class='notification is-warning'>
+          <button class='delete'></button>
+          <strong>Invalid headers</strong>
+          <p>Valid headers: #{valid_headers.join(', ')}</p>
+          <p>Current headers: #{headers.join(', ')}</p>
+        </div>"
+      }
+    end
```

Comment:
If you were to express this as JSON, perhaps something like:

```ruby
{
  invalid_headers: headers - valid_headers
  valid_headers: valid_headers
}
```

Reply:
Good idea!